### ISSUE TITLE
[Q] How do you test text-properties ?

### ISSUE THREAD

**Issue by xmaillard** [ecukes/ecukes#154]
Hello,

I am writing a new mode (sort of a chat mode).

My main chat buffer is using lui for the UI part. 

Here is what I have written so far to check for mode presence, prompt, etc.

`Feature: Entering Sententia
  In order to connect to SaT
  As a normal user
  I want to do start sententia

  Background:
    Given I switch to buffer "*sententia*"
    
  Scenario: Enter sententia mode
    When I turn on sententia-mode
    Then The current buffer should be in "sententia-mode"
    # And "sententia>" should be a prompt in buffer "*sententia*"
`

The 'And' part has been commented since I can't figure out how to test that the "sententia>" string has special text-properties. Is that just possible ?

Regards

Nota: not sure I am asking at the right place, if not, just say so.

**Comment by rejeep** [comment]
Cant you simply use `text-properties-at`? IIRC you also need to run Ecukes in [win](https://github.com/ecukes/ecukes#win) or [no win](https://github.com/ecukes/ecukes#no-win) mode because text properties are not set in batch mode.

**Comment by xmaillard** [comment]
Yes, I thought I could use `text-properties-at' but I do not know exactly.

Any help appreciated.

**Comment by rejeep** [comment]
I'm thinking something like this:

```lisp
(Then "I should see the promt"
  (lambda ()
    ;; (point-max) here is just a guess
    (should (eql (text-properties-at (point-max)) 'the-text-property))))
```

You also need to run in non-batch mode, for example

```bash
$ ecukes features/my-test.feature --no-win
```

But it's really hard to say without an understanding anything about how the project works. Do you have a link to the project?

**Comment by xmaillard** [comment]
Currently, there is no living project. I am building the tests (features, scenario, etc) and then I will build the code. As I am new to TDD/BDD, I thought it was the way to go.

I write my code "on the way"

For the UI part, I using lui-mode (from circe)

**Comment by rejeep** [comment]
Perhaps @jorgenschaefer can help out with this as I have no knowledge of the internals if lui?

**Comment by jorgenschaefer** [comment]
You can use `(goto-char lui-output-marker)` to go to the start of the prompt. Or use `(buffer-substring lui-output-marker lui-input-marker)` to get a string with properties of the full prompt.
